### PR TITLE
fix: wait for status code to be done

### PR DESCRIPTION
**Pull request by jimliu-gbd**
We thought that it's risky to get json() directly without checking the response status in check function.

### CODE REVIEW

File: decanter_ai_sdk/web_api/decanter_api.py [hunk 1]
```diff
@@ -163,25 +163,52 @@ def update_table(self, table_id, updated_column, updated_type) -> None:
...
+
+            elif task == "prediction":
+                res = self.session.get(
+                    f"{self.url}prediction/{id}", verify=False, headers=self.headers
+                )
+
+            else:
+                raise ValueError("Invalid task")
+            
+        if res.status_code == 200:
```
Comment (yuwaj6) [inline decanter_ai_sdk/web_api/decanter_api.py:189]: line# 189 ~ line# 210 這一段應該要包在while 裡面？

Comment (yuwaj6) [reply decanter_ai_sdk/web_api/decanter_api.py:189]: use res.raise_for_status() and wrap it with try-catch.

The exception for res.json should also be able to catch and retry under max-retry.

ref: https://www.geeksforgeeks.org/response-raise_for_status-python-requests/

Comment (jimliu-gbd) [reply decanter_ai_sdk/web_api/decanter_api.py:189]: > line# 189 ~ line# 210 這一段應該要包在while 裡面？

要~
太雷了沒有縮排好。

File: decanter_ai_sdk/web_api/decanter_api.py [hunk 2]
```diff
@@ -163,25 +163,54 @@ def update_table(self, table_id, updated_column, updated_type) -> None:
...
+            elif task == "prediction":
+                res = self.session.get(
+                    f"{self.url}prediction/{id}", verify=False, headers=self.headers
+                )
+            else:
+                raise ValueError("Invalid task")
+            
+            try:
+                res.raise_for_status()
+                if res.status_code == 200:
```
Comment (yuwaj6) [inline decanter_ai_sdk/web_api/decanter_api.py:190]: 有 `res.raise_for_status()`
就不用特別檢查 `if res.status_code == 200:` 囉～
大致上相同的作用

**Review by yuwaj6 — COMMENTED** [review]
LGTM to minor feedback.

File: decanter_ai_sdk/web_api/decanter_api.py (hunk 2 above)
Comment (jimliu-gbd) [reply decanter_ai_sdk/web_api/decanter_api.py:190]: 拔

**Comment by yuwaj6** [comment]
LGTM

**Pull request merged**